### PR TITLE
Add crop method to NDCubeSequence

### PR DESCRIPTION
**Pull request by DanRyanIrish**
### Description

This PR adds a `.crop` and `.crop_by_values` to `NDCubeSequence`.  It works by using the `NDCube` crop infrastructure to determine the slice items the crop inputs would require.  If not all slice items are the same, the min and max start and stop values from all cubes are used.  The slice item is them applied to the  sequence as a whole.  The sequence axis is not sliced.

Requires #472 to be merged first.

Fixes #<Issue Number>

### CODE REVIEW

File: ndcube/utils/sequence.py [hunk 1]
```diff
@@ -140,3 +142,75 @@ def cube_like_tuple_item_to_sequence_items(item, common_axis, common_axis_length
...
+            item = cube_utils.get_crop_item(lower_corner, upper_corner, wcs=wcs,
+                                            data_shape=cube.data.shape)
+        for j, s in enumerate(item):
+            starts[i, j] = s.start
+            stops[i, j] = s.stop
+    # Construct the item with which to slice the sequence from the min and max
+    # rangge of array indices determined above from all cubes.
+    starts = starts.min(axis=0)
+    stops = stops.max(axis=0)
+    return tuple([slice(None)] + [slice(start, stop) for start, stop in zip(starts, stops)])
```
Comment (DanRyanIrish) [inline ndcube/utils/sequence.py:216]:
```suggestion
    return tuple([slice(0, n_cubes)] + [slice(start, stop) for start, stop in zip(starts, stops)])
```

**Comment by DanRyanIrish** [comment]
@nabobalis this PR is now ready for a review if you are interested in giving one.

File: ndcube/utils/sequence.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 import numpy as np
 
+from ndcube.utils import cube as cube_utils
```
Comment (nabobalis) [inline ndcube/utils/sequence.py:11]: Is there a reason for this import?

File: ndcube/ndcube_sequence.py [hunk 3]
```diff
@@ -231,6 +231,144 @@ def explode_along_axis(self, axis):
...
+        """
+        Crop cubes in sequence to smallest pixel-space bounding box containing the input points.
+
+        Each input point is given as a tuple of high-level world coordinate objects.
+        This method does not crop the sequence axis.  Instead input points
+        are passed to the crop method of cubes in sequence.  Note, therefore,
+        that the input points do not include an entry for world coords only
+        associated with the sequence axis.
+        For a description of how the bounding box is defined, see the docstring
+        of the NDCube.crop method.
```
Comment (nabobalis) [inline ndcube/ndcube_sequence.py:244]: Need to check that `NDCube.crop` is linked correctly on RTD.

File: ndcube/ndcube_sequence.py [hunk 4]
```diff
@@ -231,6 +231,144 @@ def explode_along_axis(self, axis):
...
+        wcses: iterable of WCS objects or `str` (optional)
+            The WCS objects to be used to crop the cubes.
+            There must by one WCS per cube.
+            Alternatively, can be a string giving the name of the cube wcs attribute to be used,
+            namely, 'wcs', 'combined_wcs', or 'extra_coords'.
+            Default=None is equivalent to 'wcs'.
+
+        Returns
+        -------
+        : `~ndcube.NDCubeSequence`
```
Comment (nabobalis) [inline ndcube/ndcube_sequence.py:264]:
```suggestion
        `~ndcube.NDCubeSequence`
```

File: ndcube/ndcube_sequence.py [hunk 5]
```diff
@@ -231,6 +231,144 @@ def explode_along_axis(self, axis):
...
+        return self[item]
+
+    def _get_sequence_crop_item(self, *points, wcses=None, crop_by_values=False, units=None):
+        """
+        Get the slice item with which to crop an NDCubeSequence given crop inputs.
+
+        Input points are passed to the crop method of cubes in sequence.
+        Note, therefore, that the input points do not include an entry for world
+        coords only associated with the sequence axis.
+        TO learn how the bounding box is defined, see the docstrings of NDCube's
```
Comment (nabobalis) [inline ndcube/ndcube_sequence.py:315]:
```suggestion
        To learn how the bounding box is defined, see the docstrings of NDCube's
```

File: ndcube/ndcube_sequence.py [hunk 6]
```diff
@@ -231,6 +231,144 @@ def explode_along_axis(self, axis):
...
+        crop and crop_by_values methods.
```
Comment (nabobalis) [inline ndcube/ndcube_sequence.py:316]: Do these link on RTD?

**Review by nabobalis — APPROVED** [review]

File: ndcube/ndcube_sequence.py (hunk 3 above)
Comment (DanRyanIrish) [reply ndcube/ndcube_sequence.py:244]:
```suggestion
        of the `ndcube.NDCube.crop` method.
```

File: ndcube/ndcube_sequence.py [hunk 7]
```diff
@@ -231,6 +231,143 @@ def explode_along_axis(self, axis):
...
+        item = self._get_sequence_crop_item(*points, wcses=wcses)
+        return self[item]
+
+    def crop_by_values(self, lower_corner, upper_corner, units=None, wcses=None):
+        """
+        Crop cubes in sequence to smallest pixel-space bounding box containing the input points.
+
+        Each input point is given as a tuple of low-level world coordinate objects.
+        This method does not crop the sequence axis.  Instead input points
+        are passed to the crop_by_values method of cubes in sequence.  Note, therefore,
```
Comment (DanRyanIrish) [inline ndcube/ndcube_sequence.py:276]:
```suggestion
        are passed to the `ndcube.NDCube.crop_by_values` method of cubes in sequence.  Note, therefore,
```

File: ndcube/ndcube_sequence.py (hunk 6 above)
Comment (DanRyanIrish) [reply ndcube/ndcube_sequence.py:316]:
```suggestion
        `~ndcube.NDCube.crop` and `~ndcube.NDCube.crop_by_values` methods.
```

**Comment by DanRyanIrish** [comment]
@nabobalis all your comments have been addressed.  So I'll merge this once the tests are passed.  That will enable `NDCollection.crop` PR to advance.

**Pull request merged**